src/main/java/gift/service/CategoryService.java

```diff
@@ -0,0 +1,27 @@
+package gift.service;
+
+import gift.DTO.category.CategoryResponse;
+import gift.domain.Category;
+import gift.repository.CategoryRepository;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CategoryService {
+
+    private final CategoryRepository categoryRepository;
+
+    @Autowired
+    public CategoryService(CategoryRepository categoryRepository) {
+        this.categoryRepository = categoryRepository;
+    }
+
+    public List<CategoryResponse> getAllCategories() {
+        List<Category> categories = categoryRepository.findAll();
+        List<CategoryResponse> responses = categories.stream()
+                                        .map(cat -> CategoryResponse.fromEntity(cat))
+                                        .toList();
+        return responses;
+    }
```

Comment:
```suggestion
    public List<CategoryResponse> getAllCategories() {
        List<Category> categories = categoryRepository.findAll();
        return categories.stream()
                         .map(CategoryResponse::fromEntity)
                         .toList();
    }
```

처럼 메서드 참조와, 즉시 return을 통해 가독성을 높일 수도 있습니다 :)